fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/MetricUnionCompositeTableDefinition.java

```diff
@@ -113,4 +122,33 @@ public ConfigPhysicalTable build(ResourceDictionaries dictionaries, DataSourceMe
                 .filter(Objects::nonNull)
                 .collect(Collectors.toSet());
     }
+
+    /**
+     * Returns a map from availability to set of metrics.
+     *
+     * @param dictionaries  The ResourceDictionaries from which the tables are to be retrieved
+     *
+     * @return A map from <tt>Availability</tt> to set of <tt>MetricColumn</tt>
+     */
+    public Map<Availability, Set<String>> getAvailabilitiesToMetrics(ResourceDictionaries dictionaries) {
+        Set<Column> columns = buildColumns(dictionaries.getDimensionDictionary());
+        Set<String> metricNames = Utils.getSubsetByType(columns, MetricColumn.class).stream()
+                .map(MetricColumn::getName)
+                .collect(Collectors.collectingAndThen(Collectors.toSet(), ImmutableSet::copyOf));
+
+        // Construct a map of availability to its assigned metric
+        // by intersecting its underlying datasource metrics with table configured metrics
+        Map<Availability, Set<String>> availabilitiesToMetricNames = getPhysicalTables(dictionaries).stream()
```

Comment:
We should do one more thing, validate that all the metrics are being assigned to a physical table, so that we don't have a metric that is configured but ignored because none of the physical table are associated with it. Basically we can Log a warning to user that some metrics are configured but not used since no physical table support it.